### PR TITLE
Thresholds: add (optional) global threshold setting

### PR DESCRIPTION
**Pull request by ryantxu**
This adds a controller for thresholds and then uses it for overrides and for the global scope

image

### CODE REVIEW

File: src/thresholdsCtrl.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+export function polyThresholds() {
+  return {
+    controller: ThresholdsCtrl,
+    controllerAs: 'ctrl',
+    restrict: 'E',  
+    scope: {
+      thresholds: '=',
+      override: '=?',
+      onChange: '&',
```
Comment (ryantxu) [inline src/thresholdsCtrl.ts]: stumped on how to get a function passed in and then callable :(

File: src/thresholdsCtrl.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  updateThresholdColor(threshold: PolystatThreshold) {
+    console.log('WHERE DO the colors come from?', threshold);
+    // threshold.state determines the color used
+    //console.log("threshold state = " + threshold.state);
+    //console.log("override color[0]: " + override.colors[0]);
+    //console.log("override color[1]: " + override.colors[1]);
+    //console.log("override color[2]: " + override.colors[2]);
+    //threshold.color = override.colors[threshold.state];
```
Comment (ryantxu) [inline src/thresholdsCtrl.ts]: obviously missing something here!

**Comment by ryantxu** [comment]
@briangann -- do you think this is what we are after?

Apologies for making this a much bigger PR than necessary... without applying types, I am a pretty hopeless programmer!

**Comment by briangann** [comment]
looks great, merging!

**Pull request merged**